backend/open_webui/models/knowledge.py

```
@@ -89,127 +114,600 @@ class KnowledgeResponse(KnowledgeModel):
+                    return self._to_knowledge_model(result, db=db)
                 else:
                     return None
             except Exception:
                 return None
 
-    def get_knowledge_bases(self) -> list[KnowledgeUserModel]:
-        with get_db() as db:
-            knowledge_bases = []
-            for knowledge in (
+    def get_knowledge_bases(
+        self, skip: int = 0, limit: int = 30, db: Optional[Session] = None
```

Ruff (0.15.2)

[warning] Unused method argument: `skip`

(ARG002)